macaroonbakery/macaroon.py

```diff
@@ -0,0 +1,291 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+
+import base64
+import copy
+import os
+
+import bakery
+import codec
+import pymacaroons
+
+MACAROON_V1 = 1
+MACAROON_V2 = 2
+
+
+class Macaroon:
+    '''Represent an undischarged macaroon along its first
+    party caveat namespace and associated third party caveat information
+    which should be passed to the third party when discharging a caveat.
+    '''
+    def __init__(self, root_key, id, location=None,
+                 version=bakery.LATEST_BAKERY_VERSION, ns=None):
+        '''Create a new macaroon with the given root key, id and location.
+
+        If the version is more than the latest known version,
+        the latest known version will be used. The namespace is that
+        of the service creating it.
+        @param root_key bytes or string
+        @param id bytes or string
+        @param location bytes or string
+        @param version the bakery version.
+        @param ns
+        '''
+        if version > bakery.LATEST_BAKERY_VERSION:
+            version = bakery.LATEST_BAKERY_VERSION
+        # m holds the underlying macaroon.
+        self.m = pymacaroons.Macaroon(location=location, key=root_key,
```

Comment:
Does self.m need to be exposed? And does it need to be a single letter char?
Maybe s/self.m/self._macaroon/ ?